src/Features/CSharp/Portable/Completion/KeywordRecommenders/AbstractSyntacticSingleKeywordRecommender.cs

```diff
@@ -84,17 +81,16 @@ internal TestAccessor GetTestAccessor()
             public TestAccessor(AbstractSyntacticSingleKeywordRecommender recommender)
                 => _recommender = recommender;
 
-            internal async Task<IEnumerable<RecommendedKeyword>> RecommendKeywordsAsync(int position, CSharpSyntaxContext context)
+            internal async Task<ImmutableArray<RecommendedKeyword>> RecommendKeywordsAsync(int position, CSharpSyntaxContext context)
             {
                 var syntaxKind = await _recommender.RecommendKeywordAsync(position, context, CancellationToken.None).ConfigureAwait(false);
                 if (syntaxKind.HasValue)
                 {
                     var matchPriority = _recommender.ShouldPreselect(context, CancellationToken.None) ? SymbolMatchPriority.Keyword : MatchPriority.Default;
-                    return SpecializedCollections.SingletonEnumerable(
-                        new RecommendedKeyword(SyntaxFacts.GetText(syntaxKind.Value), matchPriority: matchPriority));
+                    return ImmutableArray.Create(new RecommendedKeyword(SyntaxFacts.GetText(syntaxKind.Value), matchPriority: matchPriority));
                 }
 
-                return null;
+                return ImmutableArray<RecommendedKeyword>.Empty;
```

Comment:
⚠️ This path is now allocating a Task<TResult>, where it wasn't before. Is it called often?